article/app/services/dotcomponents/ArticlePicker.scala

```diff
@@ -122,16 +122,45 @@ object ArticlePicker {
     isSupported
   }
 
-  def getTier(page: PageWithStoryPackage)(implicit request: RequestHeader): RenderType = {
+  def dcrShouldRender(request: RequestHeader): Boolean = {
+    // dcrShouldRender provides an override to let us force rendering by DCR even
+    // when an article is not supportted
+    val forceDCR = request.forceDCR
+    // our whitelist is only one article at the moment
+    val isInWhitelist = request.path == "/info/2019/dec/08/migrating-to-react"
+
+    forceDCR || isInWhitelist
+  }
 
-    val whitelistFeatures = featureWhitelist(page, request)
-    val isEnabled = conf.switches.Switches.DotcomRendering.isSwitchedOn
-    val isCommercialBetaUser = ActiveExperiments.isParticipating(DotcomRenderingAdvertisements)
+  def dcrShouldNotRender(request: RequestHeader): Boolean = {
+    val forceDCROff = request.forceDCROff
+    val dcrEnabled = conf.switches.Switches.DotcomRendering.isSwitchedOn
+
+    forceDCROff || !dcrEnabled
+  }
 
-    val tier = if ((dcrCouldRender(page, request) && isEnabled && isCommercialBetaUser && !request.forceDCROff) || request.forceDCR) RemoteRender else LocalRenderArticle
+  def getTier(page: PageWithStoryPackage)(implicit request: RequestHeader): RenderType = {
+    val whitelistFeatures = featureWhitelist(page, request)
+    val userIsInCohort = ActiveExperiments.isParticipating(DotcomRenderingAdvertisements)
+
+    // Decide if we should render this request with the DCR platform or not
+    // RemoteRender means DCR
+    val tier = if (dcrShouldNotRender(request)) {
+      // DCR is not turned on or dcr=false was passed
+      LocalRenderArticle
+    } else if (dcrShouldRender(request)) {
+      // DCR is on and either dcr=true was passed or this article is in the whitelist
+      RemoteRender
+    } else if (dcrCouldRender(page, request) && userIsInCohort) {
```

Comment:
❤️ `userIsInCohort` - really nice and descriptive.

Reply:
Nice and easy +1